pjnath/src/pjnath/ice_session.c

```diff
@@ -3734,6 +3739,56 @@ PJ_DEF(pj_status_t) pj_ice_sess_on_rx_pkt(pj_ice_sess *ice,
 
         PJ_RACE_ME(5);
 
+        if (ice->opt.check_src_addr) {
+            pj_bool_t check_addr = PJ_TRUE;
+            pj_sockaddr *raddr = &comp->rcand_check_addr;
+
+            if (!pj_sockaddr_has_addr(raddr)) {
+                for (i = 0; i < ice->rcand_cnt; ++i) {
+                    if (ice->rcand[i].comp_id == comp_id &&
+                        ice->rcand[i].checked &&
+                        pj_sockaddr_cmp(src_addr, &ice->rcand[i].addr) == 0)
+                    {
+                        /* Before ICE completed, it is still allowed to receive 
+                         * data from other candidates.
+                         */
+                        if (ice->is_complete) {
+                            pj_time_val now;
+                            pj_uint32_t duration;
+
+                            pj_gettimeofday(&now);
+                            PJ_TIME_VAL_SUB(now, ice->time_completed);
+                            duration = PJ_TIME_VAL_MSEC(now);
+
+                            if (duration > PJ_ICE_SESS_SET_RADDR_DELAY) {
+                                char psrc_addr[PJ_INET6_ADDRSTRLEN] = {0};
+
+                                if (pj_sockaddr_has_addr(src_addr)) {
+                                    pj_sockaddr_print(src_addr, psrc_addr, 
+                                                      sizeof(psrc_addr), 3);
+                                }
+                                pj_sockaddr_cp(raddr, src_addr);
+                                PJ_LOG(4, (ice->obj_name, "Using [%s] as valid"
```

Comment:
As discussed before, I don't think brackets are good idea for printing IP address because IPv6 may also add brackets. Also below.